src/Linters/Problem.hack

```diff
@@ -0,0 +1,36 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+/**
+ * A problem detected by an HHAST linter or an OCaml lint rule.
+ *
+ * Problem is more generic than LintError because Problem is associated with a
```

Comment:
the naming here is also unclear; all problems are raised by a linter, so the distinction with LintError is pretty unintuitive. It's also a bit too generic - nothing in the name corresponds to the linting framework, as apposed to the other parts of HHAST.  For example, is it a migration 'problem'?

Perhaps call this interface 'LintError', and rename that class 'SingleRuleLinterError'?

Reply:
How about `LintProblem`? The terminology problem is consistent with VS Code and other IDEs. On the other hand,  error does not align with other libraries or applications, because `error` is usually considered as a severity level in hh_client and VS Code.

Reply:
No strong feelings on what's more correct - though:
- if done, it should be consistent, i.e. all lint-related'*Error*' classes should be renamed
- if done consistently, it's a fairly big breaking change, and I'm not sure it's worth it
- if done, weak conflicting feelings:
  - the VSCode internal term is 'diagnostic'
  - IME most people call them 'lint errors', so it's my preference...
  - ... but, for now, in HHAST we report them all as WARNING severity diagnostics